micronota/bfillings/tests/test_prodigal.py

```diff
@@ -95,36 +94,37 @@ def test_base_command(self):
             c.BaseCommand,
             'cd "%s/"; %s -p single' % (getcwd(), c._command))
 
-    def test_identify_features_wrong_input(self):
+    def test_pred_wrong_input_fp(self):
+        pred = FeaturePred(None, self.tmp_dir)
         for fp in self.negative_fps:
             with self.assertRaisesRegex(
                     ApplicationError,
                     r'(Sequence read failed)|(no input sequences to analyze)'):
-                identify_features(fp, self.temp_dir, 'foo')
-
-    def test_identify_features(self):
-        for fp, params, prefix, suffix in zip(self.positive_fps,
-                                              self.positive_params,
-                                              self.positive_prefices,
-                                              self.positive_suffices):
-            res = identify_features(fp, self.temp_dir, prefix, params)
+                pred(fp)
+
+    def test_pred_run(self):
+        for fp, params, outdir in zip(self.positive_fps,
+                                      self.positive_params,
+                                      self.positive_outdir):
+            exp_d = _get_named_data_path(outdir)
+            obs_d = join(self.tmp_dir, outdir)
+            pred = FeaturePred(None, obs_d)
+            res = pred.run(fp, params)
             self.assertEqual(res['ExitStatus'], 0)
-            for i in ['-o', '-d', '-a']:
-                fp = self.get_prodigal_path('.'.join([prefix, suffix[i]]))
-                with open(fp) as f:
-                    self.assertEqual(f.read(), res[i].read())
-                res[i].close()
+            for f in listdir(exp_d):
+                self.assertTrue(cmp(join(obs_d, f), join(exp_d, f)))
```

Comment:
nope. it also checks the file content. but I forgot `shallow=False`. will
add

On Sun, Mar 13, 2016 at 1:02 PM, Jamie Morton <notifications@github.com>
wrote:

> In micronota/bfillings/tests/test_prodigal.py
> <https://github.com/biocore/micronota/pull/54#discussion_r55942761>:
>
> >              self.assertEqual(res['ExitStatus'], 0)
> > -            for i in ['-o', '-d', '-a']:
> > -                fp = self.get_prodigal_path('.'.join([prefix, suffix[i]]))
> > -                with open(fp) as f:
> > -                    self.assertEqual(f.read(), res[i].read())
> > -                res[i].close()
> > +            for f in listdir(exp_d):
> > +                self.assertTrue(cmp(join(obs_d, f), join(exp_d, f)))
>
> This is only comparing the presence/absence of files correct?
>
> It'll definitely be worthwhile to test the contents of the files.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/micronota/pull/54/files#r55942761>.
>